appinventor/appengine/src/com/google/appinventor/client/explorer/dialogs/ProjectPropertiesDialogBox.java

```diff
@@ -0,0 +1,192 @@
+        projectProperties.setCaption(Ode.MESSAGES.projectPropertiesText());
+
+        // Get current instance of YaProjectEditor
+    	projectEditor = (YaProjectEditor)Ode.getInstance().getEditorManager().getOpenProjectEditor(
+            Ode.getInstance().getCurrentYoungAndroidProjectId());
+
+        MockForm form = projectEditor.getFormFileEditor("Screen1").getForm();
+
+        // Get project properties from the screen1 MockForm
+	    EditableProperties editableProperties = form.getProperties();
+        Iterator<EditableProperty> properties = editableProperties.iterator();
+
+        // Iterate and put the editable property to the corresponding category in categoryToProperties
+        while (properties.hasNext()) {
+            EditableProperty property = properties.next();
+
+            if (!categoryToProperties.containsKey(property.getCategory())) {
+                categoryToProperties.put(property.getCategory(), new ArrayList<EditableProperty>());
+            } 
+
+            categoryToProperties.get(property.getCategory()).add(property);
+        }
+
+        // Add the Categories to ListBox - projectPropertyCategoryTitlePanel
+        for (String categoryTitle : projectPropertyCategoryTitle) {
+            projectPropertyCategoryTitlePanel.addItem(categoryTitle);
+            propertiesDeckPanel.add(getPanel(categoryTitle));
+        }
+
+        // When category is changed by the user, display related properties
+        projectPropertyCategoryTitlePanel.addChangeHandler(new ChangeHandler() {
+            @Override
+            public void onChange(ChangeEvent event) {
+                propertiesDeckPanel.showWidget(projectPropertyCategoryTitlePanel.getSelectedIndex());
+            }
+        });
+
+        projectPropertyCategoryTitlePanel.setVisibleItemCount(projectPropertyCategoryTitle.size());
+        
+        // When dialog is opened, properties related to the General category is shown
+        propertiesDeckPanel.showWidget(0);
+    }
+
+    /**
+     * Build vertical panel for each categories
+     * 
+     * @param category indicates the category for which we need to build the vertical panel
+     * @return vertical panel which contains the all the Editable Property belongs to the particualt category passed as argument
+     */
+    private VerticalPanel getPanel(String category) {
+        // Main 
+        VerticalPanel propertiesContainer = new VerticalPanel();
+        propertiesContainer.setStyleName("ode-propertyDialogVerticalPanel");
+
+        ArrayList<EditableProperty> properties = categoryToProperties.get(category);
+
+        for (EditableProperty property : properties) {
+            // container for displaing one editable property
+            VerticalPanel propertyContainer = new VerticalPanel();
+            propertyContainer.setStyleName("ode-propertyDialogPropertyContainer");
+
+            // name of the EditableProperty
+            Label name = new Label(property.getName());
```

Comment:
We should make sure this internationalizes correctly.

Reply:
can you please provide me some reference or example for this ?

Reply:
You will want to call `ComponentsTranslation.getPropertyName(property.getName())`